packages/core/src/browser/widgets/react-widget.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as ReactDOM from "react-dom";
+import { injectable } from "inversify";
+import { DisposableCollection } from "../../common";
+import { BaseWidget, Message } from "./widget";
+import { ReactElement } from "react";
+
+@injectable()
+export class ReactWidget extends BaseWidget {
```

Comment:
Please add a deprecation note to `VirtualWidget` and put that `ReactWidget` should be used instead.